buildlib/az-helpers.sh

```diff
@@ -247,3 +257,149 @@ get_arch() {
         echo "$arch"
     fi
 }
+
+declare -A COLORS=(
+    [RED]='\033[0;31m'
+    [GREEN]='\033[0;32m'
+    [BLUE]='\033[0;34m'
+    [CYAN]='\033[0;36m'
+    [NC]='\033[0m' # No Color
+)
+
+function log_script_info() {
+    echo -e "${COLORS[CYAN]}Script Information${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+    echo -e "${COLORS[GREEN]}Script:${COLORS[NC]} $0"
+    echo -e "${COLORS[GREEN]}Working Directory:${COLORS[NC]} $(pwd)"
+    echo -e "${COLORS[GREEN]}User:${COLORS[NC]} $(whoami)"
+}
+
+function log_machine_info() {
+    echo -e "\n${COLORS[CYAN]}Machine Information${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+    if [ -f /.dockerenv ]; then
+        echo -e "${COLORS[GREEN]}Running in container:${COLORS[NC]} Yes"
+        echo -e "${COLORS[GREEN]}Agent Machine Name:${COLORS[NC]} $AGENT_MACHINENAME"
+        echo -e "${COLORS[GREEN]}Container Image:${COLORS[NC]} $(awk -F= '/^PRETTY_NAME=/ {gsub(/"/, "", $2); print $2}' /etc/os-release)"
```

Comment:
Is that possible to somehow also obtain docker image path here or maybe even show the full command to run the docker manually?

Something like:
```
Container image path: rdmz-harbor.rdmz.labs.mlnx/ucx/centos8-mofed23-cuda11_asan:1
CMD to run container: docker run --rm -ti -v `pwd`:`pwd` -w `pwd` --cap-add=SYS_PTRACE -v /hpc/local:/hpc/local -v /auto/sw_tools:/auto/sw_tools --gpus all --net=host   --ulimit memlock=-1:-1 --device=/dev/infiniband/  rdmz-harbor.rdmz.labs.mlnx/ucx/centos8-mofed23-cuda11_asan:1 /bin/bash

```

Reply:
that's exactly what I'm trying to add, I may have to add an environment variable

Reply:
Added an environment variable for it